controllers/servers/host_definer/watcher/watcher_helper.py

```diff
@@ -0,0 +1,263 @@
+            self.patch_host_definition(host_definition_manifest)
+        else:
+            logger.info('Creating host Definition: {}'.format(host_definition.name))
+            self._create_host_definition(host_definition_manifest)
+        self.set_host_definition_status(host_definition.name, phase)
+
+    def get_host_definition_name(self, management_address, node_name):
+        return '{0}.{1}'.format(management_address, node_name).replace('_', '.')
+
+    def get_host_definition_manifest(self, host_definition):
+        return {
+            'apiVersion': settings.CSI_IBM_API_VERSION,
+            'kind': settings.HOST_DEFINITION_KIND,
+            'metadata': {
+                'name': host_definition.name,
+            },
+            'spec': {
+                'hostDefinition': {
+                    settings.MANAGEMENT_ADDRESS_FIELD: host_definition.management_address,
+                    settings.NODE_NAME_FIELD: host_definition.node_name,
+                    settings.NODE_ID_FIELD: host_definition.node_id,
+                    settings.SECRET_NAME_FIELD: host_definition.secret_name,
+                    settings.SECRET_NAMESPACE_FIELD: host_definition.secret_namespace,
+                },
+            },
+        }
+
+    def _get_prefix(self):
+        return os.getenv('PREFIX')
+
+    def get_connectivity(self):
+        return os.getenv('CONNECTIVITY')
+
+    def undefine_host_and_host_definition_with_events(self, host_definition):
+        node_name = host_definition.node_name
+        logger.info('Verifying that host {} is undefined from storage {}'.format(node_name,
+                                                                                 host_definition.management_address))
+        response = self.undefine_host_and_host_definition(host_definition)
+        if response.error_message:
+            self.set_host_definition_status(host_definition.name, settings.PENDING_DELETION_PHASE)
+            self._create_event_to_host_definition(host_definition.name, response.error_message)
+
+    def _create_event_to_host_definition(self, host_definition_name, message):
+        host_definition, _ = self._get_host_definition(host_definition_name)
+        if host_definition:
+            self.add_event_to_host_definition(host_definition, message)
+
+    def get_node_name_from_node_id(self, node_id):
+        node_name, _, _, _ = utils.get_node_id_info(node_id)
+        return node_name
+
+    def add_event_to_host_definition(self, host_definition, message):
+        logger.info('Creating event for host definition: {} error event: {}'.format(host_definition.name, message))
+        event = self._get_event_for_object(host_definition, message)
+        self.create_event(settings.DEFAULT_NAMESPACE, event)
+
+    def is_host_can_be_defined(self, node_name):
+        if self.is_dynamic_node_labeling_allowed():
+            return True
+        return self.is_node_has_managed_by_host_definer_label(node_name)
```

Comment:
```suggestion
        return self.is_dynamic_node_labeling_allowed() or self.is_node_has_managed_by_host_definer_label(node_name)
```

Reply:
if we are using `_is_node_has_managed_by_host_definer_label` every time that `is_dynamic_node_labeling_allowed `is not `True `so the OR statement is what we want ( if `is_dynamic_node_labeling_allowed `is `True `the or wont check `is_node_has_managed_by_host_definer_label`)